engine/class_modules/monk/sc_monk.cpp

```diff
@@ -608,22 +624,26 @@ struct monk_heal_t : public monk_action_t<heal_t>
     {
       player_t* t = ( execute_state ) ? execute_state->target : target;
 
-      if ( td( t )->dots.enveloping_mist->is_ticking() )
+      if ( auto td = find_td( t ) )
```

Comment:
I'm fairly sure you can end up in a situation where no target data has been created but a DoT is still ticking. That is, application of a DoT does not force the creation of target data for that target.

Reply:
I would hope that whoever is applying the dot on the target will create the target-data through a get_target_data call.
Thus the assumption that if no target data can be found, there was never a dot applied and it can't be ticking.

Any reason why that should not be the case?

Reply:
DoTs need not be in target data. In fact, most of them aren't. Take a look at action_t::trigger_dot, no target data in sight.

Reply:
Yeah you are absolutely right. I thought they did call player_t::get_target_data at some point, but they don't.

I'll adjust this then.

Reply:
I guess the other option would be to let action_t::get_dot create target data as well when initially creating the dot.
Not sure if that would be better worse performance-wise.

The main idea of using find_target_data is that you do not have to create the objects for all player<>target combinations in the first place. (Which isn't that important anymore with profilesets anyway)

Reply:
I would just leave it as is. There are plenty of actors that have dots but no debuffs to create and plenty of situations that trigger dots and don't need target data (such as dots from trinkets on debuff-light classes).